sdk/src/main/java/io/opentelemetry/sdk/resources/Resource.java

```diff
@@ -116,7 +117,45 @@ public static Resource getTelemetrySdk() {
    */
   public static Resource create(Attributes attributes) {
     checkAttributes(Objects.requireNonNull(attributes, "attributes"));
-    return new AutoValue_Resource(attributes);
+    Attributes.Builder attrBuilder = Attributes.newBuilder();
+    attributes.forEach(new NullValueFilter(attrBuilder));
+    return new AutoValue_Resource(attrBuilder.build());
+  }
+
+  private static final class NullValueFilter implements KeyValueConsumer<AttributeValue> {
+    private final Attributes.Builder attrBuilder;
+
+    private NullValueFilter(Attributes.Builder attrBuilder) {
+      this.attrBuilder = attrBuilder;
+    }
+
+    @Override
+    public void consume(String key, AttributeValue value) {
+      boolean ignore = false;
+      switch (value.getType()) {
+        case STRING:
+          ignore = value.getStringValue() == null;
+          break;
+        case LONG_ARRAY:
+          ignore = value.getLongArrayValue().equals(Collections.<Long>emptyList());
```

Comment:
It might be easier just to call `isEmpty()` rather than comparing explicitly to an empty list. And, this makes me think we could have an `isEmpty()` method on the AttributeValue, and then you can skip all of the switch branches and have the logic much more simple.